server/proxy/pf_capture.c

```diff
@@ -0,0 +1,77 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol Implementation
+ * FreeRDP Proxy Server Session Capture
+ *
+ * Copyright 2019 Kobi Mizrachi <kmizrachi18@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdio.h>
+#include <string.h>
+
+#include "pf_capture.h"
+
+static BOOL pf_capture_create_dir_if_not_exists(const char* path)
+{
+	if (PathFileExistsA(path))
+		return TRUE;
+
+	return CreateDirectoryA(path, NULL);
+}
+
+/* creates a directory to store captured session frames.
+ *
+ * @context: current session.
+ *
+ * directory path will be: base_dir/username/session-start-date.
+ *
+ * it is important to call this function only after the connection is fully established, as it uses
+ * settings->Username and settings->ServerHostname values to create the directory. After the
+ * connection established, we know that those values are valid.
+ */
+BOOL pf_capture_create_session_directory(pClientContext* pc)
+{
+	proxyConfig* config = pc->pdata->config;
+	rdpSettings* settings = pc->context.settings;
+	SYSTEMTIME localTime;
+	char tmp[MAX_PATH];
```

Comment:
Might be a good idea to remove use of `MAX_PATH` as this is just the size of a single path segment (and under certain circumstances not even that is true)
`sprintf` functions have the neat property that they return the length required if called with `NULL` ;)

Reply:
hmm, so you're suggesting to call sprintf with NULL, get the size, allocate a buffer with this size and then call sprintf again?

Reply:
and what about the usage of MAX_PATH in pf_capture_save_frame?

Reply:
correct, yes.
this is of course true for every use. (I know we still use it in some places in freerdp, but that is one thing to take care of, introducing it in new code is no good idea)